File: guides/spring-boot-to-micronaut-consuming-rest/micronautframework/java/src/main/java/example/micronaut/Application.java
package example.micronaut;

import io.micronaut.context.event.ApplicationEventListener;
import io.micronaut.context.event.StartupEvent;
import io.micronaut.runtime.Micronaut;
import io.micronaut.scheduling.annotation.Async;
import jakarta.inject.Inject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Application implements ApplicationEventListener<StartupEvent> {  // <1>
    private static final Logger LOG = LoggerFactory.getLogger(Application.class);
    public static void main(String[] args) {
        Micronaut.run(Application.class, args);
    }

    @Inject // <2>
    QuotersClient quotersClient;

    @Override
    public void onApplicationEvent(StartupEvent event) {  // <1>
        logQuote();
    }

    @Async // <3>
    void logQuote() {
        Quote quote = quotersClient.random();
        LOG.info("{}", quote);
    }
}

File: guides/micronaut-scheduled/python/src/example/micronaut/application.py
import time

from jakarta.inject import Singleton
from micronaut.runtime.server.event import ServerStartupEvent
from micronaut.runtime.event.annotation import EventListener

from .register_use_case import RegisterUseCase


@Singleton  # <1>
class Application:  # <2>
    def __init__(self, register_use_case: RegisterUseCase):  # <3>
        self.register_use_case = register_use_case

    @EventListener
    def onApplicationEvent(self, event: ServerStartupEvent) -> None:  # <4>
        self.register_use_case.register("harry@micronaut.example")
        time.sleep(20)
        self.register_use_case.register("ron@micronaut.example")
